src/libmeasurement_kit/ooni/utils.cpp

```diff
@@ -23,14 +23,46 @@ void resolver_lookup(Callback<Error, std::string> callback, Settings settings,
     resolver_lookup_impl(callback, settings, reactor, logger);
 }
 
-IPLocation::IPLocation(std::string path_country_, std::string path_asn_,
+/* static */ Var<IpLocationProxy> IpLocationProxy::global() {
+    static Var<IpLocationProxy> singleton(new IpLocationProxy);
+    return singleton;
+}
+
+Var<IpLocation> IpLocationProxy::open(std::string country, std::string asn,
```

Comment:
I find it a bit misleading that this method name is called `open`. I would expect that an open method would actually create open handles to the various geoip files, but instead what it does is that it just creates an instance of `IPLocation`. The files are actually opened lazily when you first attempt to resolve something.

At the minimum I would add a line stating that this is the behaviour.